packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt

```diff
@@ -6,65 +6,63 @@ import android.os.Build
 import com.google.android.gms.common.GooglePlayServicesUtil
 import com.google.android.play.core.review.ReviewManager
 import com.google.android.play.core.review.ReviewManagerFactory
-import expo.modules.core.ExportedModule
-import expo.modules.core.ModuleRegistry
-import expo.modules.core.Promise
-import expo.modules.core.interfaces.ActivityProvider
-import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.functions.Queues
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 
-class StoreReviewModule(private val mContext: Context) :
-  ExportedModule(mContext) {
-  companion object {
-    private const val NAME = "ExpoStoreReview"
-  }
+class StoreReviewModule : Module() {
+  private val context: Context
+    get() = requireNotNull(appContext.reactContext) { "React Application Context is null" }
 
-  private lateinit var mActivityProvider: ActivityProvider
+  private val currentActivity
+    get() = appContext.activityProvider?.currentActivity
+      ?: throw MissingCurrentActivityException()
 
-  override fun getName(): String {
-    return NAME
-  }
+  override fun definition() = ModuleDefinition {
+    Name("ExpoStoreReview")
+
+    AsyncFunction("isAvailableAsync") { promise: Promise ->
+      isAvailableAsync(promise)
+    }
 
-  override fun onCreate(moduleRegistry: ModuleRegistry) {
-    mActivityProvider = moduleRegistry.getModule(ActivityProvider::class.java)
+    AsyncFunction("requestReview") { promise: Promise ->
+      requestReview(promise)
+    }.runOnQueue(Queues.MAIN)
```

Comment:
Does it have to be called on the main thread? I don't see anything similar in the previous implementation.

Reply:
When you're using the `AsyncFuncton` component, the provided function is always dispatched on a different thread than the one owned by the JS. So, when you don't provide `runOnQueue(Queues.MAIN)`, your block will be run on a background thread - it won't block the js side. I assume that on iOS that function has to be called from the UI thread which doesn't seem to be needed on Android.

Reply:
Yeah, I can confirm that these functions must be called from the UI thread on iOS, otherwise they could crash in some circumstances 😉

Reply:
Ok, that makes sense. Thanks, @lukmccall. I knew both platforms shouldn't be treated the same so don't know what I was thinking 😅 I'll make those changes